python/lsst/sphgeom/box3d.cc

```diff
@@ -0,0 +1,183 @@
+/*
+ * LSST Data Management System
+ * See COPYRIGHT file at the top of the source tree.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+#include "pybind11/pybind11.h"
+
+#include <memory>
+
+#include "lsst/sphgeom/Box3d.h"
+#include "lsst/sphgeom/python/relationship.h"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace lsst {
+namespace sphgeom {
+namespace {
+
+// Convert a Python integer to a C++ int in [-3, 3). If the Python integer
+// is out of bounds, an IndexError is raised.
```

Comment:
You're right - I'll fix the comment and move the utility function into a header file.